### PR TITLE
WIP: Base Card Component

### PR DESCRIPTION
**Pull request by gilad215**
**What**:
Creating the Base Card components. I'm still waiting for @priley86 response on #192 on how to approach the other cards , but so far I created the following base components: `Card` `Card Container` `Card Body` `Card Heading` `Card Title` `Card Footer`

**Link to Storybook**:
https://gilad215.github.io/patternfly-react/

Cards -> Base Card

**Additional issues**:
Storybook is showing API of unrelated classes (e.g. Grid,Col,Row,Button), I saw this issue on other components as well so i guess this is known.

**TODO:**
- Add Tests
- You tell me?

### CODE REVIEW

File: src/components/Filter/Filter.js [hunk 1]
```diff
@@ -5,7 +5,8 @@ import { getContext } from 'recompose';
 import { toolbarContextTypes } from '../Toolbar/ToolbarConstants';
 
 // Disabled eslint due to `isDescendantOfToolbar` being a context property we don't want passed by consumers
-const Filter = ({ children, className, isDescendantOfToolbar, ...rest }) => { // eslint-disable-line
```
Comment (gilad215) [inline src/components/Filter/Filter.js]: I didn't do this, i think its Prettier

**Comment by priley86** [comment]
@gilad215 thanks very much for starting this component! I will review this one more shortly w/ you...

**Comment by priley86** [comment]
for starters... branch name is `master`... and you now you have my local all out of whack 😸 ... i will just `git fetch gilad215 master:cards` instead 🌞 ... do you mind using a new branch name next time?

also too - we will need to ensure the *first* commit is run with commitizen syntax... so `npm run commit` can help you there...

**Comment by gilad215** [comment]
@priley86 Thanks for the tips, ill work with `npm run commit` from now on.
I committed another component since your comment, This component `CardTimeFrameFilter` is debatable i would say, please share your thoughts on it.

Any feedback is appreciated, Thanks.

File: src/components/Cards/CardBody.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import ClassNames from 'classnames';
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const CardBody = ({ children, className, ...props }) => {
+  const classes = ClassNames('p', 'card-pf-body', className);
+
+  return (
+    <p className={classes} {...props}>
```
Comment (priley86) [inline src/components/Cards/CardBody.js]: should render a `div` instead i believe... the paragraph can follow in children...

File: src/components/Cards/Card.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  /** Child nodes */
+  children: PropTypes.node.isRequired,
+  /** Additional element css classes */
+  className: PropTypes.string,
+  /** Accent & Aggregate bool */
+  accented: PropTypes.bool,
+  aggregated: PropTypes.bool
+};
+
+Card.defaultProps = {
```
Comment (priley86) [inline src/components/Cards/Card.js]: these aren't required by current linter (so default is already false) ;)

File: src/components/Cards/Card.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
+import ClassNames from 'classnames';
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const Card = ({ children, className, accented, aggregated, ...props }) => {
+  const classes = ClassNames(
```
Comment (priley86) [inline src/components/Cards/Card.js]: same thing as (and more consistent w/ the others here):  ```const classes = ClassNames(
    'card-pf',
    {
      'card-pf-accented': accented,
      'card-pf-aggregate-status': aggregated
    },
    className
  );```

File: src/components/Cards/CardTitle.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
+import ClassNames from 'classnames';
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const CardTitle = ({ children, className, ...props }) => {
+  const classes = ClassNames('h2', 'card-pf-title', className);
+
+  return (
+    <h4 className={classes} {...props}>
```
Comment (priley86) [inline src/components/Cards/CardTitle.js]: why render an `h4` with `.h2` class? Can't we just render an `h2`?

File: src/components/Cards/Cards.stories.js [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+
+  let story = (
+    <Grid>
+      <Row style={{ marginBottom: '20px' }}>
+        <Col xs={6} sm={4} md={4}>
+          <CardContainer>
+            <Card accented={accentedBool} aggregated={aggregatedBool}>
+              <CardHeading>
+                <CardTimeFrameFilter>
+                  <DropdownButton
```
Comment (priley86) [inline src/components/Cards/Cards.stories.js]: let's go ahead and create a `CardDropdownButton` if you don't mind...

File: src/components/Cards/CardLinkWithIcon.js [hunk 7]
```diff
@@ -0,0 +1,26 @@
+import ClassNames from 'classnames';
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const CardLinkWithIcon = ({ href, children, className, ...props }) => {
```
Comment (priley86) [inline src/components/Cards/CardLinkWithIcon.js]: my preference would be to rename it `CardLink` instead and use the same component for rendering "Footer link" (link w/o icon) and "Add New Cluster" (link w/ icon as children).

File: src/components/Cards/CardLinkWithIcon.js [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+  const classes = ClassNames('card-pf-link-with-icon', className);
+
+  return (
+    <div>
+      <p>
+        <a href={href} className={classes} {...props}>
```
Comment (priley86) [inline src/components/Cards/CardLinkWithIcon.js]: we should ensure it supports `onClick` as well as `href` i believe.

File: src/components/Cards/CardLinkWithIcon.js [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+    </div>
+  );
+};
+CardLinkWithIcon.propTypes = {
+  /** Child node - contents of the element */
+  children: PropTypes.node.isRequired,
+  /** Additional element css classes */
+  className: PropTypes.string,
+
+  href: PropTypes.string
```
Comment (priley86) [inline src/components/Cards/CardLinkWithIcon.js]: please ensure you have docgen comments on all propTypes:
```
/** href definition */
href: PropTypes.string
```

File: src/components/Cards/CardTimeFrameFilter.js [hunk 10]
```diff
@@ -0,0 +1,20 @@
+import ClassNames from 'classnames';
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const CardTimeFrameFilter = ({ children, className, ...props }) => {
+  const classes = ClassNames('dropdown card-pf-time-frame-filter', className);
```
Comment (priley86) [inline src/components/Cards/CardTimeFrameFilter.js]: This is another case of a custom button group it seems... we just want to remove the `bsClass` (should not have to render a `div`, let the `ButtonGroup` do it for you): 
```
 const CustomButtonGroup = props => {
    return <ButtonGroup {...props} bsClass=" " />;
  };
```

Comment (priley86) [reply src/components/Cards/CardTimeFrameFilter.js]: (full example from before was here: https://github.com/patternfly/patternfly-react/blob/master/src/components/Table/TableDropdownKebab.js )

**Comment by priley86** [comment]
thanks again for the pr @gilad215 and welcome to the community 👍 🥇 

I think maybe starting a fresh branch may be easier (but leave it up to you). Should not need any changes to package-lock (so please ensure you are branching up the latest master branch as first step). Also, `out` directory seems to be getting checked in (but it is not necessary).

I think after reviewing the Card issues some more, we can probably just target the bottom 4 most card rows in this example for the "base card" issue you are tackling first (all of the card patterns *after* the trend /utilization trend cards... we just want to tackle the patterns with the empty "[card content]" as a first step I think):
https://rawgit.com/patternfly/patternfly/master-dist/dist/tests/cards.html

I would try to update your Storybook and split out the card examples exactly as they are rendered there (and additionally lay out your tests this way as well). It will be easiest to compare 1 to 1 this way I think...

does it make sense? I am happy to give you a PR if you are getting stuck... 

good luck!

**Comment by gilad215** [comment]
Thank you @priley86 for your feedback, it seems i can't change the target branch, so i'm creating a new PR.

**Pull request closed without merging**